### PR TITLE
[Owners review]Added 'custom_close' tag to InitExtCal

### PR DESCRIPTION
**Pull request by Akshata440**
### What does this Pull Request accomplish?

- Added a 'custom-close' tag to InitExtCal in functions.py
- Made changes in sevice_helpers.mako so that functions with 'custom_close' tag can be autogenerated.
- Recorded all the changes made to metadata in CHANGES.md

### Why should this Pull Request be merged?
The cleanup lambda for InitExtClose can now be autogenerated to use CloseExtCal.

### What testing has been done?

I was able to autogenerate InitExtCal.

### CODE REVIEW

**Review by doshirohan — APPROVED** [review]

File: source/codegen/metadata/nidmm/functions.py [hunk 1]
```diff
@@ -1624,7 +1624,7 @@
     },
     "InitExtCal": {
         "init_method" : True,
-        "codegen_method" : None,
+        "custom_close" : "CloseExtCal(id,NIDMM_EXTCAL_ACTION_ABORT)",
```
Comment (jonlugoatx) [inline source/codegen/metadata/nidmm/functions.py]: Put a space after the `,`:
```suggestion
        "custom_close" : "CloseExtCal(id, NIDMM_EXTCAL_ACTION_ABORT)",
```

File: source/codegen/templates/service_helpers.mako [hunk 2]
```diff
@@ -19,7 +20,11 @@ ${initialize_input_params(function_name, parameters)}
       };
       uint32_t session_id = 0;
       const std::string& session_name = request->session_name();
+      %if 'custom_close' in function_data:
+      auto cleanup_lambda = [&] (uint32_t id) { library_->${function_data['custom_close']}; };
+      %else:
       auto cleanup_lambda = [&] (uint32_t id) { library_->${config['close_function']}(id); };
+      %endif
```
Comment (jonlugoatx) [inline source/codegen/templates/service_helpers.mako]: Consider factoring out the close call into a python variable in the `<% %>` block above:
```python
close_function_call = function_data['custom_close'] if 'custom_close' in function_data else f"{config['close_function']}(id)"
```
```cpp
auto cleanup_lambda = [&] (uint32_t id) { library_->${close_function_call}; };
```
```

File: source/codegen/metadata/nidmm/functions.py (hunk 1 above)
Comment (reckenro) [inline source/codegen/metadata/nidmm/functions.py]: [Minor] Can you add a space between the comma and second argument?

**Review by jonlugoatx — APPROVED** [review]

**Review by reckenro — APPROVED** [review]

File: source/codegen/templates/service_helpers.mako (hunk 2 above)
Comment (Akshata440) [reply source/codegen/templates/service_helpers.mako]: Yes, changed it.

File: source/codegen/metadata/nidmm/functions.py (hunk 1 above)
Comment (Akshata440) [reply source/codegen/metadata/nidmm/functions.py]: Done

Comment (Akshata440) [reply source/codegen/metadata/nidmm/functions.py]: Done

**Pull request merged**